cloudos_cli/import_wf/import_wf.py

```diff
@@ -0,0 +1,128 @@
+from abc import ABC, abstractmethod
+from urllib.parse import urlsplit
+import requests
+from cloudos_cli.utils.errors import BadRequestException
+from cloudos_cli.utils.requests import  retry_requests_post
+import json
+
+
+class WFImport(ABC):
+    def __init__(self, cloudos_url, cloudos_apikey, workspace_id, platform,
+                 workflow_name, workflow_url, workflow_docs_link="", main_file=None, verify=True):
+        self.cloudos_url = cloudos_url
+        self.workflow_url = workflow_url.rstrip('.git')
+        self.workspace_id = workspace_id
+        self.platform = platform
+        self.parsed_url = urlsplit(self.workflow_url)
+        self.main_file = main_file
+        self.repo_name = ""
+        self.repo_owner = ""
+        self.repo_host = f"{self.parsed_url.scheme}://{self.parsed_url.netloc}"
+        self.headers = {
+            "Content-Type": "application/json",
+            "apikey": cloudos_apikey
+        }
+        self.payload = {
+            "workflowType": "nextflow",
+            "repository": {
+                "platform": platform,
+                "repositoryId": None,
+                "name": None,
+                "owner": {
+                    "login": None,
+                    "id": None
+                },
+                "isPrivate": True,
+                "url": self.workflow_url,
+                "commit": "",
+                "branch": ""
+            },
+            "name": workflow_name,
+            "description": "",
+            "isPublic": False,
+            "mainFile": "main.nf",
+            "defaultContainer": None,
+            "processes": [],
+            "docsLink": workflow_docs_link,
+            "team": workspace_id
+        }
+        self.get_repo_url = ""
+        self.get_repo_params = dict()
+        self.get_repo_main_file_url = ""
+        self.get_repo_main_file_params = ""
+        self.post_request_url = f"{cloudos_url}/api/v1/workflows?teamId={workspace_id}"
+        self.verify = verify
+
+    def get_repo_main_file(self):
+        repo_owner_urlencode = self.repo_owner.replace("/", "%2F")
+        get_repo_main_file_url = f"{self.cloudos_url}/api/v1/git/{self.platform}/getWorkflowConfig/{self.repo_name}/{repo_owner_urlencode}"
+        get_repo_main_file_params = dict(host=self.repo_host, teamId=self.workspace_id)
+        r = requests.get(get_repo_main_file_url, params=get_repo_main_file_params, headers=self.headers)
+        r_data = r.json()
```

Comment:
@juan-lifebit here, you are not implementing any error handling, so it is very difficult to see when things fail from the server side. Could you please implement:
- Our retry wrapper for get ( from https://github.com/lifebit-ai/cloudos-cli/blob/main/cloudos_cli/utils/requests.py)
- Some kind of error strategy that at least captures API response?, like https://github.com/lifebit-ai/cloudos-cli/blob/b6162aecf8848eecbce14b986d3a9a0ad6de3b2c/cloudos_cli/clos.py#L59-L63